qiskit/opflow/converters/z2_symmetries.py

```diff
@@ -0,0 +1,418 @@
+
+        self._symmetries = symmetries
+        self._sq_paulis = sq_paulis
+        self._sq_list = sq_list
+        self._tapering_values = tapering_values
+
+    @property
+    def symmetries(self):
+        """ return symmetries """
+        return self._symmetries
+
+    @property
+    def sq_paulis(self):
+        """ returns sq paulis """
+        return self._sq_paulis
+
+    @property
+    def cliffords(self) -> List[PauliSumOp]:
+        """
+        Get clifford operators, build based on symmetries and single-qubit X.
+        Returns:
+            a list of unitaries used to diagonalize the Hamiltonian.
+        """
+        cliffords = [
+            (PauliOp(pauli_symm) + PauliOp(sq_pauli)) / np.sqrt(2)
+            for pauli_symm, sq_pauli in zip(self._symmetries, self._sq_paulis)
+        ]
+        return cliffords
+
+    @property
+    def sq_list(self):
+        """ returns sq list """
+        return self._sq_list
+
+    @property
+    def tapering_values(self):
+        """ returns tapering values """
+        return self._tapering_values
+
+    @tapering_values.setter
+    def tapering_values(self, new_value):
+        """ set tapering values """
+        self._tapering_values = new_value
+
+    def __str__(self):
+        ret = ["Z2 symmetries:"]
+        ret.append("Symmetries:")
+        for symmetry in self._symmetries:
+            ret.append(symmetry.to_label())
+        ret.append("Single-Qubit Pauli X:")
+        for x in self._sq_paulis:
+            ret.append(x.to_label())
+        ret.append("Cliffords:")
+        for c in self.cliffords:
+            ret.append(str(c))
+        ret.append("Qubit index:")
+        ret.append(str(self._sq_list))
+        ret.append("Tapering values:")
+        if self._tapering_values is None:
+            possible_values = [
+                str(list(coeff))
+                for coeff in itertools.product([1, -1], repeat=len(self._sq_list))
+            ]
+            possible_values = ', '.join(x for x in possible_values)
+            ret.append("  - Possible values: " + possible_values)
+        else:
+            ret.append(str(self._tapering_values))
+
+        ret = "\n".join(ret)
+        return ret
+
+    def copy(self) -> "Z2Symmetries":
+        """
+        Get a copy of self.
+        Returns:
+            copy
+        """
+        return deepcopy(self)
+
+    def is_empty(self) -> bool:
+        """
+        Check the z2_symmetries is empty or not.
+        Returns:
+            Empty or not
+        """
+        return self._symmetries == [] or self._sq_paulis == [] or self._sq_list == []
+
+    @classmethod
+    def find_Z2_symmetries(  # pylint: disable=invalid-name
+            cls, operator: PauliSumOp
+    ) -> 'Z2Symmetries':
+        """
+        Finds Z2 Pauli-type symmetries of an Operator.
+        Returns:
+            a z2_symmetries object contains symmetries,
+            single-qubit X, single-qubit list.
```

Comment:
I think you can connect these two lines of comments. Sphinx seems to cause an error related around here.
```
/home/vsts/work/1/s/qiskit/opflow/converters/z2_symmetries.py:docstring of qiskit.opflow.converters.Z2Symmetries.find_Z2_symmetries:4:Unexpected indentation.
ERROR: InvocationError for command /home/vsts/work/1/s/.tox/docs/bin/sphinx-build -W -b html -j auto docs/ docs/_build/html (exited with code 2)
```

Reply:
Thanks. After fixing this error, another error related to `legacy/Z2Symmetries` occured. So I will fix this after #5596.